### PR TITLE
feat(openapi): add an initial tutorial

### PR DESCRIPTION
**Pull request by sennyeya**
## Hey, I just made a Pull Request!

Created an initial tutorial on how to onboard to the OpenAPI tools.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by Parsifal-M — COMMENTED** [review]
Nice! All seems quite clear quite looking forward to trying it out 👏 😄

**Comment by benjdlambert** [comment]
@sennyeya this is looking good! I wonder if you wanted to add a sidebar item for this? Maybe under the `plugins` section and just link to this open API docs part? Something like `Creating plugins with OpenAPI (Experimental)` or something?

File: docs/openapi/01-getting-started.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+1. `@backstage/repo-tools`, this package contains all OpenAPI related commands for your plugins. We will be using this throughout the tutorial.
+2. `@opticdev/optic`, this package is a dependency of `@backstage/repo-tools` but is only required for OpenAPI related commands.
+
+You should install both of the above packages in the _root_ of your workspace.
+
+## Storing your OpenAPI specification
+
+You should create a new folder, `src/schema` in your backend plugin to store your OpenAPI (and any other) specifications. For example, if you're adding a specification to the catalog plugin, you would add a `src/schema` folder to `plugins/catalog-backend`, making a `plugins/catalog-backend/src/schema` directory. This directory should have an `openapi.yaml` file inside.
+
+> Currently, only the `.yaml` ending is supported, not `.yml`.
```
[the remark this answers is not in the document]
Comment (sennyeya) [reply docs/openapi/01-getting-started.md]: Makes sense, updated in https://github.com/backstage/backstage/pull/21794/commits/dcd782bdc8e36329c2bea2f0b338a36e8fc84d3f

**Comment by sennyeya** [comment]
@benjdlambert Added as a new category under plugins with the existing docs. I expect to write some more documentation on how to write OpenAPI specs and wanted a little bit more room to house it (and any others that I may create).

File: microsite/sidebars.json [hunk 2]
```diff
@@ -248,6 +248,15 @@
...
       "plugins/feature-flags",
+      {
+        "type": "category",
+        "label": "OpenAPI",
+        "items": [
+          "openapi/01-getting-started",
+          "openapi/generate-client",
+          "openapi/test-case-validation"
+        ]
+      },
```
Comment (tudi2d) [inline microsite/sidebars.json:259]: I believe you additionally have to add the category to the `mkdocs.yml` - like e.g. for "Backends and APIs" (https://github.com/backstage/backstage/blob/master/mkdocs.yml#L131-L135)

Missed it also before - bit confusing - hope that helps! :)

**Review by tudi2d — CHANGES_REQUESTED** [review]
Thanks for contributing this & great to see your name her - hope you are doing ok! Small addition for the sidebar needed.

File: microsite/sidebars.json (hunk 2 above)
Comment (sennyeya) [reply microsite/sidebars.json:259]: Updated! Is there an easy way to test that locally? I was getting weird yarn errors with the existing `techdocs-cli` command.

**Review by benjdlambert — APPROVED** [review]

**Comment by sennyeya** [comment]
@tudi2d Can you take another look at this? I updated the mkdocs config as well.

**Comment by awanlin** [comment]
If you rebase that should fix the E2E tests ;)

File: microsite/sidebars.json (hunk 2 above)
Comment (tudi2d) [reply microsite/sidebars.json:259]: I would expect that you can test it by running the local dev of the microsite. Not sure about the `techdocs-cli` errors :/

**Review by tudi2d — APPROVED** [review]
thanks @sennyeya - lgtm :shipit:

**Pull request merged**